lighthouse-core/computed/trace-of-tab.js

```diff
@@ -43,7 +36,20 @@ class TraceOfTab {
    * @return {Promise<LH.Artifacts.TraceOfTab>}
   */
   static async compute_(trace) {
-    return LHTraceProcessor.computeTraceOfTab(trace);
+    const traceOfTab = await LHTraceProcessor.computeTraceOfTab(trace);
+    const {timings, timestamps, firstContentfulPaintEvt} = traceOfTab;
+    const {firstContentfulPaint: firstContentfulPaintTiming} = timings;
+    const {firstContentfulPaint: firstContentfulPaintTs} = timestamps;
+    if (!firstContentfulPaintEvt || !firstContentfulPaintTiming || !firstContentfulPaintTs) {
+      throw new LHError(LHError.errors.NO_FCP);
+    }
+
+    return {
+      ...traceOfTab,
+      firstContentfulPaintEvt,
+      timings: {...timings, firstContentfulPaint: firstContentfulPaintTiming},
+      timestamps: {...timestamps, firstContentfulPaint: firstContentfulPaintTs},
+    };
```

Comment:
ha, I don't entirely trust tsc with object spreading + composition, but any change I make here gives the right error, so good job compiler! (and @patrickhulce)